lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1702,6 +1702,7 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+		$params['filter[posts_per_page]'] = $params['per_page'];
```

Comment:
In favor—:+1: for removing `posts_per_page`. My understanding of `filter` is that it's intended as a shim for query modifications not formally supported by first-order query params; since we have `per_page`, omitting the filter param feels appropriate.